### ISSUE TITLE
QR code as NFT visual?

### ISSUE THREAD

**Issue by mjohnson518** [mjohnson518/ChainShot_Final_Project#10]
Kicking around the idea of generating a qr code based on the url of the ipfs cid to act as the visual component of the erc-721's.

QR = https://gateway.ipfs.io/ipfs/${CID.path}

**Comment by brossetti1** [comment]
potential api for this

https://goqr.me/api/

**Comment by mjohnson518** [comment]
Need to make a get call to http://api.qrserver.com/v1/create-qr-code/?data=%7Binsert_url_here%7D&size=100x100

@Spazzaroth are you looking to tackle this?

**Comment by MrDeadCe11** [comment]
yeah, as soon as we have the ipfs address we can plug it into that api and get the qr code.  can even change the size of it etc.

**Comment by brossetti1** [comment]
was looking at the docs and realized i could grab all the params easily - just adding this here for reference:

# https://api.qrserver.com/v1/create-qr-code

param_data
param_size
param_charsetsource
param_charsettarget
param_ecc
param_color
param_bgcolor
param_margin
param_qzone

param_data - https://goqr.me/api/doc/create-qr-code/#param_data
param_size - https://goqr.me/api/doc/create-qr-code/#param_size
param_charsetsource - https://goqr.me/api/doc/create-qr-code/#param_charsetsource
param_charsettarget - https://goqr.me/api/doc/create-qr-code/#param_charsettarget
param_ecc - https://goqr.me/api/doc/create-qr-code/#param_ecc
param_color - https://goqr.me/api/doc/create-qr-code/#param_color
param_bgcolor - https://goqr.me/api/doc/create-qr-code/#param_bgcolor
param_margin - https://goqr.me/api/doc/create-qr-code/#param_margin
param_qzone - https://goqr.me/api/doc/create-qr-code/#param_qzone

# https://api.qrserver.com/v1/read-qr-code

param_fileurl
param_file
param_outputformat

param_data - https://goqr.me/api/doc/read-qr-code/#param_fileurl
param_size - https://goqr.me/api/doc/read-qr-code/#param_file
param_charsetsource - https://goqr.me/api/doc/read-qr-code/#param_outputformat

**Comment by brossetti1** [comment]
not sure if this is useful but looks useful

https://github.com/ethereum/EIPs/issues/945

[youtube demo](https://www.youtube.com/watch?v=XRrsrFofEic)
[implementation](https://cipher-qr-scan.netlify.app/)
